check_powerdns_rec.py

```diff
@@ -61,26 +59,43 @@ def parse_args():
 
     parser = argparse.ArgumentParser(
         prog=__plugin_name__,
-        description='Icinga/Nagios plugin, interned to check PowerDNS status using pdns_control. '
+        description='Icinga/Nagios plugin, interned to check PowerDNS status using either rec_control or the API.'
+                    'rec_control is the default interface to obtain statistisc'
                     'A non-zero exit code is generated, if the numbers of DNS queries per seconds exceeds'
                     ' warning/critical values. Additionally the plugin checks for the security-status of PowerDNS. ',
         epilog='This program is free software: you can redistribute it and/or modify '
                'it under the terms of the GNU General Public License as published by '
                'the Free Software Foundation, either version 3 of the License, or '
                'at your option) any later version. Author: ' + __author__)
-    parser.add_argument('-S', '--socket-dir', help='Where the PowerDNS controlsocket will live', type=str, default='')
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('-A', '--api-host', help='PowerDNS API host (do not combine with --socket-dir or --test)', type=str)
+    group.add_argument('-T', '--test', help='Test case; Use fake data (do not combine with --api-host or --socket-dir)',
+                        action='store_true')
+    group.add_argument('-S', '--socket-dir', help='PDNS Control tool Socket directory (do not combine with --socket-dir or --test)', type=str, default='')
+
+    parser.add_argument('-P', '--api-port', help='PowerDNS API port (default 8082)', type=int, default=8082)
+    parser.add_argument('-k', '--api-key', help='PowerDNS API key', type=str, default='')
     parser.add_argument('-n', '--config-name', help='Name of PowerDNS virtual configuration', type=str, default='')
```

Comment:
I'd say `--config-name` is mandatory, if "mode" is socket. Maybe default should be `None` instead?